Lib/test/pythoninfo.py

```diff
@@ -731,9 +731,12 @@ def collect_windows(info_add):
 
     import subprocess
     try:
+        # When wmic.exe output is redirected to a pipe,
+        # it uses the OEM code page
         proc = subprocess.Popen(["wmic", "os", "get", "Caption,Version", "/value"],
                                 stdout=subprocess.PIPE,
                                 stderr=subprocess.PIPE,
+                                encoding="oem",
```

Comment:
Some of Microsoft's tools support formatting output as XML or JSON, which can generally be parsed more reliably. For example:

```
import subprocess
import xml.etree.ElementTree as ET

p = subprocess.run('wmic.exe os get Caption,Version /format:rawxml', 
        capture_output=True, encoding='oem')
root = ET.fromstring(p.stdout)
caption = root.find('.//PROPERTY[@NAME="Caption"]/VALUE').text
version = root.find('.//PROPERTY[@NAME="Version"]/VALUE').text
```

Reply:
Please propose a PR if you prefer to parse XML than plain text. I don't want to invest too much time in this code, since I'm not using Windows :-)

Reply:
Well, maybe a more "correct" implementation is required if the function enters the stdlib platform module, but this PR is only about test.pythoninfo which is only used ... for tests ;-)